packages/world/src/modules/core/CoreModule.sol

```diff
@@ -0,0 +1,37 @@
+// SPDX-License-Identifier: MIT
+pragma solidity >=0.8.0;
+
+import { ROOT_NAMESPACE } from "../../constants.sol";
+import { WithMsgSender } from "../../WithMsgSender.sol";
+
+import { IModule } from "../../interfaces/IModule.sol";
+
+import { NamespaceOwner } from "../../tables/NamespaceOwner.sol";
+import { ResourceAccess } from "../../tables/ResourceAccess.sol";
+import { Systems } from "../../tables/Systems.sol";
+import { FunctionSelectors } from "../../tables/FunctionSelectors.sol";
+
+/**
+ * The CoreModule registers internal World tables.
+
+ * Note:
+ * This module is required to be delegatecalled via `World.registerRootSystem`.
+ * because otherwise the registration of tables would require the `registerTable`
+ * function selector to already exist on the World, but it is only
+ * added in `RegistrationModule`, which is installed after `CoreModule`.
+ */
```

Comment:
ooc do we have a way to detect if a given module has been installed, aside from looking at each individual thing being registered?

If it's worth the gas, might be cool to have as a way to introspect the world and provide nicer errors (either in client or in Solidity) if something reverts because a module was not installed before another one.

Reply:
We could add a `Modules` table that stores which modules are installed in a world and a "dependency check" in the `install` method to revert if a required module is not found! One point of friction is it would require modules to have a unique name (at least unique per namespace), but maybe that's worth it for the benefit of better developer experience.